src/pinterestScraper.py

```diff
@@ -4,130 +4,122 @@
+            self.driver.execute_script(f"window.scrollTo(0, {Y})")  # Scroll down the page
             sleep(1)
+            # Store the link of each image if the page contains the targeted images
             try:
                 container = self.driver.find_element_by_xpath('//div[@data-test-id="grid"]//div[@class="vbI XiG"]')
                 image_list = container.find_elements_by_xpath('//div[@class="Yl- MIw Hb7"]//img')
-                # ('//div[@data-test-id="grid"]//div[2]//img')
-                print(len(image_list))
+                print(f"Number of images successfully extracted: {len(image_list)}")
                 self.image_set.update([link.get_attribute('src') for link in image_list])
-                print(len(self.image_set), 'image set')
+                print(f"Number of uniques images: {len(self.image_set)}")
             except: 
-                print('Some error, likely no pics')
-        # sleep(5)
-        # self.driver.refresh()
-        # sleep(5)ß
-        # image_list = self.driver.find_elements_by_xpath('//div[@data-test-id="grid"]//div[2]/a')
-        # self.links = []
-        # for i, item in enumerate(category_list):
-        #     self.links.append(item.get_attribute('href'))
-        # print(self.links)
+                print('Some errors occurred, most likely due to no images present')
 
     def get_image_source(self, link: str) -> None:
+        """Move to the image source page
+
+        Args
+        ---------------------
+        link: str
+        """
         self.driver.get(link)
         sleep(0.5)
-        # self.src = link.get_attribute('src')
         self.src = link
 
-    def download_images(self, i) -> None:
-        # filename = self.category.split('/')[0]
-        urllib.request.urlretrieve(self.src, f"../{self.save_path}/{self.save_path}_{i}.jpg")
+    def download_images(self, i: int) -> None:
+        """Download the image
+        Args
+        ---------------------
+        i: int
+        """
+        urllib.request.urlretrieve(self.src, 
+                                f"{self.root_save_path}/{self.save_path}/{self.save_path}_{i}.jpg")
 
     def get_category_images(self) -> None:
+        """Run all the required functions to automate image retrival 
+        """
         self.get_category_links()
         self.create_folders()
+        # Loop through each category
         for category in self.category_link_list:
             self.category = category.replace('https://www.pinterest.co.uk/ideas/', "")
             self.extract_links()
             self.save_path = f'{self.category.split("/")[0]}'
             for i, link in enumerate(self.image_set):
-                if '75x75' not in link:
+                if '75x75' not in link: # Download image if it is not a profile picture
                     self.get_image_source(link)
                     self.download_images(i)
             self.image_set = set()
-        self.driver.quit()
```

Comment:
Maybe try and split this function into two functions since the loops are nested, could become computationally expensive.